src/main/java/io/github/thebusybiscuit/slimefun4/core/services/sounds/SoundEffect.java

```diff
@@ -0,0 +1,183 @@
+    EXPLOSIVE_BOW_HIT_SOUND(Sound.ENTITY_GENERIC_EXPLODE, 1F, 1F),
+    FISHERMAN_ANDROID_FISHING_SOUND(Sound.ENTITY_PLAYER_SPLASH, 0.3F, 0.7F),
+    FLASK_OF_KNOWLEDGE_FILLUP_SOUND(Sound.ENTITY_EXPERIENCE_ORB_PICKUP, 1F, 0.5F),
+    INFUSED_HOPPER_TELEPORT_SOUND(Sound.ENTITY_ENDERMAN_TELEPORT, 0.5F, 2F),
+    INFUSED_MAGNET_TELEPORT_SOUND(Sound.ENTITY_ENDERMAN_TELEPORT, 0.25F, 0.9F),
+    IRON_GOLEM_ASSEMBLER_ASSEMBLE_SOUND("entity.iron_golem.repair", 0.5F, 1F),
+    JETBOOTS_THRUST_SOUND(Sound.ENTITY_TNT_PRIMED, 0.25F, 1F),
+    JETPACK_THRUST_SOUND(Sound.ENTITY_GENERIC_EXPLODE, 0.25F, 1F),
+    JUICER_USE_SOUND(Sound.ENTITY_PLAYER_SPLASH, 1F, 1F),
+    MAGICAL_EYE_OF_ENDER_USE_SOUND(Sound.ENTITY_ENDERMAN_TELEPORT, 1F, 1F),
+    MAGIC_SUGAR_CONSUME_SOUND(Sound.ENTITY_GENERIC_EAT, 1F, 1F),
+    PLAYER_RESEARCHING_SOUND(Sound.ENTITY_BAT_TAKEOFF, 0.7F, 1F),
+    PORTABLE_CRAFTER_OPEN_SOUND(Sound.BLOCK_WOODEN_BUTTON_CLICK_ON, 1F, 1F),
+    PRESSURE_CHAMBER_FINISH_SOUND(Sound.ENTITY_ARROW_HIT_PLAYER, 1F, 1F),
+    PRESSURE_CHAMBER_WORKING_SOUND(Sound.ENTITY_TNT_PRIMED, 1F, 1F),
+    SLIME_BOOTS_FALL_SOUND(Sound.BLOCK_SLIME_BLOCK_FALL, 1F, 1F),
+    SMELTERY_CRAFT_SOUND(Sound.BLOCK_LAVA_POP, 1F, 1F),
+    SOULBOUND_RUNE_RITUAL_SOUND(Sound.ENTITY_GENERIC_EXPLODE, 0.3F, 1F),
+    SPLINT_CONSUME_SOUND(Sound.ENTITY_SKELETON_HURT, 1F, 1F),
+    TOME_OF_KNOWLEDGE_USE_SOUND(Sound.ENTITY_PLAYER_LEVELUP, 1F, 1F),
+    TRASH_CAN_OPEN_SOUND(Sound.BLOCK_ANVIL_LAND, 1F, 1F),
+    VAMPIRE_BLADE_HEALING_SOUND(Sound.ENTITY_ARROW_HIT_PLAYER, 0.7F, 0.7F),
+    VILLAGER_RUNE_TRANSFORM_SOUND(Sound.ENTITY_VILLAGER_CELEBRATE, 1F, 1.4F),
+    VITAMINS_CONSUME_SOUND(Sound.ENTITY_GENERIC_EAT, 1F, 1F),
+    WIND_STAFF_USE_SOUND(Sound.ENTITY_TNT_PRIMED, 1F, 1F);
+
+    private final String defaultSound;
+    private final float defaultVolume;
+    private final float defaultPitch;
+
+    SoundEffect(@Nonnull String sound, float volume, float pitch) {
+        Validate.notNull(sound, "The Sound id cannot be null!");
+        Validate.isTrue(volume >= 0, "The volume cannot be a negative number.");
+        Validate.isTrue(pitch >= 0.5, "A pitch below 0.5 has no effect on the sound.");
+
+        this.defaultSound = sound;
+        this.defaultVolume = volume;
+        this.defaultPitch = pitch;
+    }
+
+    SoundEffect(@Nonnull Sound sound, float volume, float pitch) {
+        Validate.notNull(sound, "The Sound id cannot be null!");
+        Validate.isTrue(volume >= 0, "The volume cannot be a negative number.");
+        Validate.isTrue(pitch >= 0.5, "A pitch below 0.5 has no effect on the sound.");
+
+        /*
+         * Only Minecraft 1.16+ implements Keyed for Sound.
+         * So we need to check this first.
+         */
+        if (sound instanceof Keyed) {
+            this.defaultSound = sound.getKey().getKey();
+        } else {
+            this.defaultSound = sound.name().toLowerCase(Locale.ROOT).replace('_', '.');
+        }
```

Comment:
we only support 1.16+ so this isnt necessary, tho I think thats already covered in the pr to this one